docs/reference/server-events.md

```diff
@@ -0,0 +1,982 @@
+</AposCodeBlock>
+
+### `afterConvert`
+
+Triggered after page data is run through the schema module's `convert` method, validating its data to the page type's field schema.
+
+#### Parameters
+
+- `req`: The active request
+- `data`: The data being saved to the page, prior to schema conversion
+- `page`: The page as it will be saved, following schema conversion
+
+<AposCodeBlock>
+  ```javascript
+  handlers(self, options) {
+    return {
+      'afterConvert': {
+        async handlerName(req, data, page) { ... }
+      }
+    };
+  }
+  ```
+  <template v-slot:caption>
+    modules/@apostrophecms/page/index.js
+  </template>
+</AposCodeBlock>
+
+### `beforeMove`
+
+Triggered just *before* a page is moved from one position in the page tree to another.
+
+#### Parameters
+
+- `req`: The active request
+- `page`: The page being moved
+- `target`: The page used as a target for positioning in the page tree
+- `position`: The position to place the page in relation to the target
+
+<AposCodeBlock>
+  ```javascript
+  handlers(self, options) {
+    return {
+      'beforeMove': {
+        async handlerName(req, page, target, position) { ... }
+      }
+    };
+  }
+  ```
+  <template v-slot:caption>
+    modules/@apostrophecms/page/index.js
+  </template>
+</AposCodeBlock>
+
+### `afterMove`
+
+Triggered just *after* a page is moved from one position in the page tree to another.
+
+#### Parameters
+
+- `req`: The active request
+- `page`: The page being moved
+- `data`: An object containing the following properties:
+  - `originalSlug`: The page's slug prior to moving
+  - `originalPath`: The page's path prior to moving
+  - `changed`: An array of all pages that were changed by the move, including children of the moved page
+  - `target`: The page used as a target for positioning in the page tree
+  - `position`: The position to place the page in relation to the target
+
+<AposCodeBlock>
+  ```javascript
+  handlers(self, options) {
+    return {
+      'afterMove': {
+        async handlerName(req, page, data) { ... }
+      }
+    };
+  }
+  ```
+  <template v-slot:caption>
+    modules/@apostrophecms/page/index.js
+  </template>
+</AposCodeBlock>
+
+### `serveQuery`
+
+Triggered just before the database query executes to find the best matching page for a request. This can be used to make final adjustments to the query.
+
+#### Parameters
+
+- `req`: The active request
+- `query`: The database query that will execute
+
+<AposCodeBlock>
+  ```javascript
+  handlers(self, options) {
+    return {
+      'serveQuery': {
+        async handlerName(req, query) { ... }
+      }
+    };
+  }
+  ```
+  <template v-slot:caption>
+    modules/@apostrophecms/page/index.js
+  </template>
+</AposCodeBlock>
+
+### `serve`
+
+Triggered just before a requested page is served.
+
+#### Parameters
+
+- `req`: The active request
+
+<AposCodeBlock>
+  ```javascript
+  handlers(self, options) {
+    return {
+      'serve': {
+        async handlerName(req) { ... }
```

Comment:
I'm hoping we can evolve later into having short concrete examples for these, as this is pretty mysterious otherwise.